src/qibo/models/circuit.py

```diff
@@ -1210,6 +1210,17 @@ def from_qasm(cls, qasm_code, accelerators=None, density_matrix=False):
         parser = QASMParser()
         return parser.to_circuit(qasm_code, accelerators, density_matrix)
 
+    def to_qir(self):
+        """Convert circuit to QIR string."""
+        try:
+            from qbraid.transpiler.conversions.qasm2 import qasm2_to_qasm3
+            from qbraid.transpiler.conversions.qasm3 import qasm3_to_pyqir
+        except ModuleNotFoundError as e:
+            raise ModuleNotFoundError(
+                "The optional dependency qbraid is missing, please install it with `poetry install --with qir`"
+            ) from e
```

Comment:
*Note*: The `qasm3_to_pyqir` function requires that both `qbraid` and `qbraid-qir` are installed. Or equivalently, the `qbraid[qir]` extra:
- https://github.com/qBraid/qBraid/blob/f1e05d59e79f90c153937af3e32b45a10bdfb67e/pyproject.toml#L56
- https://github.com/qBraid/qBraid/blob/f1e05d59e79f90c153937af3e32b45a10bdfb67e/qbraid/transpiler/conversions/qasm3/qasm3_extras.py#L63